examples/decoding/plot_mixed_gambles_space_net.py

```diff
@@ -0,0 +1,46 @@
+"""
+SpaceNet on Jimura et al "mixed gambles" dataset.
+==================================================
+
+The segmenting power of SpaceNet is quite visible here.
+"""
+# author: DOHMATOB Elvis Dopgima,
+#         GRAMFORT Alexandre
+
+
+### Load data ################################################################
+import numpy as np
+import nibabel
+from scipy import ndimage
+from nilearn.datasets import fetch_mixed_gambles
+data = fetch_mixed_gambles(n_subjects=16, return_raw_data=False)
+zmaps, object_category, mask_img = data.zmaps, data.gain, data.mask_img
+
+
+### Fit TV-L1 #################################################################
+from nilearn.decoding import SpaceNetRegressor
+decoder = SpaceNetRegressor(mask=mask_img, penalty="tv-l1",
+                            eps=1e-1,  # prefer large alphas
+                            memory="cache", memory_level=3)
+decoder.fit(zmaps, object_category)  # fit
+
+### Visualize TV-L1 weights
+import matplotlib.pyplot as plt
+from nilearn.plotting import plot_stat_map
+from nilearn.image import mean_img
+plot_stat_map(mean_img(decoder.coef_img_), title="tv-l1", display_mode="yz",
```

Comment:
OK, this doesn't look right. It's probably for historical reasons. Lemme rm
the mean_img stuff and rerun.

On Sat, Jul 18, 2015 at 5:49 PM, Gael Varoquaux <notifications@github.com>
wrote:

> In examples/decoding/plot_mixed_gambles_space_net.py
> <https://github.com/nilearn/nilearn/pull/657#discussion_r34948870>:
>
> > +data = fetch_mixed_gambles(n_subjects=16, return_raw_data=False)
> > +zmaps, object_category, mask_img = data.zmaps, data.gain, data.mask_img
> > +
> > +
> > +### Fit TV-L1 #################################################################
> > +from nilearn.decoding import SpaceNetRegressor
> > +decoder = SpaceNetRegressor(mask=mask_img, penalty="tv-l1",
> > +                            eps=1e-1,  # prefer large alphas
> > +                            memory="cache", memory_level=3)
> > +decoder.fit(zmaps, object_category)  # fit
> > +
> > +### Visualize TV-L1 weights
> > +import matplotlib.pyplot as plt
> > +from nilearn.plotting import plot_stat_map
> > +from nilearn.image import mean_img
> > +plot_stat_map(mean_img(decoder.coef_img_), title="tv-l1", display_mode="yz",
>
> I don't understand why you need to use 'mean_img' here?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nilearn/nilearn/pull/657/files#r34948870>.
>



-- 
DED